### PR TITLE
planner: fix index merge doesn't take effect when using prefix key (#20425)

### LINKED ISSUE #16407: Index merge doesn't take effect when using prefix key
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
```
create table t(a int,b char(100),key(a),key(b(10));
desc select /*+ use_index_merge(t) */ * from t where a=10 or b="x";
```
### 2. What did you expect to see?
use_index_merge takes effect
### 3. What did you see instead?
```
+-------------------------+----------+-----------+---------------+-------------------------------------+
| id                      | estRows  | task      | access object | operator info                       |
+-------------------------+----------+-----------+---------------+-------------------------------------+
| TableReader_7           | 8000.00  | root      |               | data:Selection_6                    |
| └─Selection_6           | 8000.00  | cop[tikv] |               | or(eq(d1.t.a, 10), eq(d1.t.b, "x")) |
|   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo      |
+-------------------------+----------+-----------+---------------+-------------------------------------+
3 rows in set, 1 warning (0.01 sec)
```
### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
Release Version: v4.0.0-rc-33-gd69868961
Git Commit Hash: d69868961361126d4788b68fd6cd597adb3ef66c
Git Branch: release-4.0
UTC Build Time: 2020-04-15 02:59:00
GoVersion: go1.13.8
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #20425 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #16407 

Problem Summary:

Index merge doesn't take effect when using prefix key.

### What is changed and how it works?

The condition `tableFilter` is not empty is too strict.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- None

### Release note 

- None

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -1130,14 +1130,84 @@ func (s *testIntegrationSerialSuite) TestIssue16837(c *C) {
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create table t(a int,b int,c int,d int,e int,unique key idx_ab(a,b),unique key(c),unique key(d))")
 	tk.MustQuery("explain select /*+ use_index_merge(t,c,idx_ab) */ * from t where a = 1 or (e = 1 and c = 1)").Check(testkit.Rows(
+<<<<<<< HEAD
```
Comment (qw4990) [inline planner/core/integration_test.go:1133]: @tangwz Please fix conflicts.

**Review by eurekaka — COMMENTED** [review]
LGTM

**Review by time-and-fate — COMMENTED** [review]
LGTM

**Review by SunRunAway — APPROVED** [review]
LGTM

**Pull request merged**